sentry_sdk/hub.py

```diff
@@ -222,17 +190,28 @@ def __init__(
                 scope = copy.copy(other_scope)
         else:
             client = client_or_hub
+            Scope.get_global_scope().set_client(client)
+
         if scope is None:
             scope = Scope()
 
         self._stack = [(client, scope)]
         self._last_event_id = None  # type: Optional[str]
         self._old_hubs = []  # type: List[Hub]
 
+        # This forked isolation scope if the hub is used as a context manager
+        self._scope = Scope.get_isolation_scope().fork()
+        self._old_scopes = []  # type: List[Scope]
+
     def __enter__(self):
         # type: () -> Hub
         self._old_hubs.append(Hub.current)
         _local.set(self)
+
+        isolation_scope = Scope.get_isolation_scope()
+        self._old_scopes.append(isolation_scope)
+        scope._isolation_scope.set(self._scope)
```

Comment:
Where does `scope` come from from?

Reply:
`scope` is the module the `_isolation_scope` ContextVar is defined in.

Reply:
But we don't import it anywhere? I'm confused

Reply:
Ahh it's at the end of the file. Nevermind.